### PR TITLE
Convert mod_vcard to gen_hook

### PR DESCRIPTION
**Pull request by kamilwaz**
Convert mod_vcard to gen_hook

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/147899) / small_tests / ff5d26901ae947f19c8596939989fe9f27e1d0bc
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3833/147899/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3833/147899/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-10-28_13.02.00/lib.mongooseim.logs/run.2022-10-28_13.02.01/suite.log.html)

File: src/vcard/mod_vcard.erl [hunk 1]
```diff
@@ -458,42 +456,46 @@ unsafe_set_vcard(HostType, From, VCARD) ->
...
+    {ok, Acc}.
 
-%% #rh
-remove_user(Acc, User, Server) ->
+-spec remove_user(Acc, Params, Extra) -> {ok, Acc} when
+    Acc :: mongoose_acc:t(),
+    Params :: map(),
+    Extra :: map().
+remove_user(Acc, #{jid := #jid{luser = User, lserver = Server}}, #{host_type := HostType}) ->
     HostType = mongoose_acc:host_type(Acc),
```
Comment (pawlooss1) [inline src/vcard/mod_vcard.erl:494]: I think we can remove this line since HostType is already bound in function's head.

**Review by pawlooss1 — COMMENTED** [review]
Just one minor comment.

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/148346) / small_tests / 5d4df61642cb363d0afbea2a6a4cef1c16d3a82a
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3833/148346/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3833/148346/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-11-02_10.29.42/lib.mongooseim.logs/run.2022-11-02_10.29.43/suite.log.html)

**Review by pawlooss1 — APPROVED** [review]

**Pull request merged**